compiler/rustc_session/src/options.rs

```diff
@@ -1284,6 +1284,8 @@ options! {
         "choose which RELRO level to use"),
     remap_cwd_prefix: Option<PathBuf> = (None, parse_opt_pathbuf, [TRACKED],
         "remap paths under the current working directory to this path prefix"),
+    restrict_vtable_fn_ptrs: bool = (false, parse_bool, [TRACKED],
+        "restrict the targets of virtual table function pointer calls"),
```

Comment:
Is there a strong reason to even make this a flag? Can we emit these files unconditionally?

Reply:
The main reasons are that currently emitting restrictions currently (1) requires wrapper functions at ever vtable call, because CBMC uses fragile naming conventions for specifying the function call site, and (2) the restrictions could be incomplete if the Rust code is doing certain types of casts that we don't handle, meaning the restriction set would be too small.

The 3rd even more transient reason is that the CBMC PR has not landed, so restrictions fail on our minimum required CBMC version.

Reply:
Could we get that broken into an issue, with e.g. links to CBMC issues or PRs we depend on?

This change is awesome, and I'd like to make sure we have a well-lit path to making it the always-on default. e.g. do we know what the "certain types of casts that we don't handle" are, and can we document that?

(This is obviously not a blocker to merging, I just don't want it accidentally dropped or anything!)